speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -339,43 +374,112 @@ static void OnActivate(void* /* dummy */)
 static void SelectFile(void* /* dummy */)
 {
     // Opens a file dialog on Windows
-    // Create file dialog
+
+    // Initialize COM interface
     HRESULT hresult = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
     if (FAILED(hresult)) { return; }
+
+    // Create a file dialog
     IFileDialog* fileDialog;
-    COMDLG_FILTERSPEC filter[1] = {{L"Executables", L"*.exe"}};
     hresult = CoCreateInstance(CLSID_FileOpenDialog, NULL, CLSCTX_ALL, IID_IFileDialog, reinterpret_cast<void**>(&fileDialog));
-    if (FAILED(hresult)) { CoUninitialize(); return; }
-    // Open file dialog
+    if (FAILED(hresult)) 
+    {
+    	CoUninitialize();
+    	return;
+    }
+
+    // Set file types for file dialog, then open it
+    COMDLG_FILTERSPEC filter[1] = { {L"Executables", L"*.exe"} };
     hresult = fileDialog->SetFileTypes(1, filter);
-    if (FAILED(hresult)) { fileDialog->Release(); CoUninitialize(); return; }
+    if (FAILED(hresult))
+    {
+	    fileDialog->Release();
+    	CoUninitialize();
+    	return;
+    }
     hresult = fileDialog->Show(NULL);
-    if (FAILED(hresult)) { fileDialog->Release(); CoUninitialize(); return; }
+    if (FAILED(hresult))
+    {
+	    fileDialog->Release();
+    	CoUninitialize();
+    	return;
+    }
+
     // Get filename
     IShellItem* shellItem;
     hresult = fileDialog->GetResult(&shellItem);
-    if (FAILED(hresult)) { fileDialog->Release(); CoUninitialize(); return; } // I feel like I should release shellItem here as well, but the Microsoft Docs does not do so at this stage
+    if (FAILED(hresult))
+    {
+	    fileDialog->Release();
+    	CoUninitialize();
+    	return;
+    } // I feel like I should release shellItem here as well, but the Microsoft Docs do not do so at this stage
```

Comment:
You repeat this code so much it is probably a good candidate for a function/macro